Implement customized reset functionality

EpicYoshiMaster opened the pull request:
## Changes
- Uses patch file for localized text
- Fixes an issue where combined events did not pass the action through for recursive checks
- Implements two new options for whether pressing New Game or starting a Preset will reset splits
- Adds a hotkey which can do this manually at any time for runs that use several files / want to start from a non-preset location.

EpicYoshiMaster committed:
Implement customized reset functionality

2767mr commented on timer/utils.js:
```diff
@@ -2,10 +2,6 @@ import { Hook } from './hooks.js';
-		ig.lang.labels.sc.gui.options['dontResetTimerOnDeath'] = {name: 'Don\'t reset timer on death', description: 'Don\'t reset timer on death. \\c[1]WARNING: This will affect the actual IGT!'};
-		ig.lang.labels.sc.gui.options['printEvents'] = {name: 'Print all events', description: 'Print all possible events that can be split on. Use "Log level: Default"'};
-		ig.lang.labels.sc.gui.options['roomTimer'] = {name: 'Display room timer', description: 'Displays a room timer'};
-		ig.lang.labels.sc.gui.options.headers['ccTimer'] = 'CCTimer';
```
The reason this was done in code is because it applies to all languages. If I use the CCTimer as it is in the PR and set my game to German it looks like this: 
<img width="350" alt="image" src="https://user-images.githubusercontent.com/2082553/215267306-96dec9e3-3579-4f8a-92d0-f80ceb990ff0.png">

2767mr commented on timer/utils.js:
```diff
@@ -26,9 +22,33 @@ export class Utils {
+			resetSplitsPress() {
+				return ig.input.pressed("reset-splits");
+			}
+		});
```
I usually discourage this because there is nothing in the game that is affected by this inject. It also makes to code less readable because when you try to find where this is defined you have to look through literally the entire mods codebase manually.
That said, this implementation is pretty clean and it's obvious what it does so I'll allow it.

2767mr commented on timer/plugin.js:
```diff
@@ -35,6 +35,20 @@ export default class CCTimer extends Plugin {
 	}
 
+	prestart() {
+		//This will not work if defined in main, so it's placed here instead
+		sc.OPTIONS_DEFINITION["keys-reset-splits"] = {
+			cat: sc.OPTION_CATEGORY.CONTROLS,
+			hasDivider: true,
+			header: 'ccTimer',
+			init: {
+				key1: ig.KEY.SEMICOLON,
+				key2: void 0
+			},
+			type: 'CONTROLS',
+		};
```
It also doesn't work if there is no input-api. Either add a check for it like in the Jetpack mod or a warning in the displayed description.

2767mr reviewed (changes requested):
I approved it by mistake. Please resolve the issues above first.

EpicYoshiMaster committed:
Address requested changes

- labels moved back into utils
- removed injection into sc.Control
- input-api is now an optional dependency

(also fixed a visual timer bug that I forgot to address a while back)
